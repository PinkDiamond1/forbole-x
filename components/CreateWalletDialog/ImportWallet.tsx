import { Button, DialogActions, DialogContent, Typography, Box, TextField } from '@material-ui/core'
import useTranslation from 'next-translate/useTranslation'
import React from 'react'
import useStyles from './styles'

interface ImportWalletProps {
  onConfirm(password: string): void
}

const ImportWallet: React.FC<ImportWalletProps> = ({ onConfirm }) => {
  const { t } = useTranslation('common')
  const classes = useStyles()
  const [name, setName] = React.useState('')

  return (
    <>
      <DialogContent className={classes.dialogContent}>
        <Typography>{t('moniker')}</Typography>
        <TextField
          fullWidth
          variant="filled"
          placeholder={t('wallet name')}
          value={name}
          onChange={(e) => setName(e.target.value)}
          InputProps={{
            disableUnderline: true,
          }}
        />
        <Box mb={30} />
      </DialogContent>
      <DialogActions>
        <Box flex={1} display="flex" flexDirection="column" mb={3}>
          <Button
            className={classes.button}
            variant="contained"
            color="primary"
            disabled={!name}
            onClick={() => onConfirm(name)}
          >
            {t('import')}
          </Button>
        </Box>
      </DialogActions>
    </>
  )
}

export default ImportWallet
