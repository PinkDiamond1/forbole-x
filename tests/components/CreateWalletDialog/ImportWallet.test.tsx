import React from 'react'
import renderer from 'react-test-renderer'
import ImportWallet from '../../../components/CreateWalletDialog/ImportWallet'

const onConfirm = jest.fn()

describe('component: CreateWalletDialog - ImportWallet', () => {
  it('renders correctly', () => {
    const component = renderer.create(<ImportWallet onConfirm={onConfirm} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('renders correctly with wallet name', () => {
    const component = renderer.create(<ImportWallet onConfirm={onConfirm} />)
    renderer.act(() => {
      component.root.findByType('input').props.onChange({
        target: {
          value: 'wallet',
        },
      })
    })
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  it('calls onConfirm when button is clicked', () => {
    const component = renderer.create(<ImportWallet onConfirm={onConfirm} />)
    renderer.act(() => {
      component.root.findByType('input').props.onChange({
        target: {
          value: 'wallet',
        },
      })
    })
    renderer.act(() => {
      component.root.findByType('button').props.onClick()
    })
    expect(onConfirm).toBeCalledWith('wallet')
  })
})

afterEach(() => {
  jest.clearAllMocks()
})
