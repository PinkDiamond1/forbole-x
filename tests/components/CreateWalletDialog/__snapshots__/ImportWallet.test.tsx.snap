// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: CreateWalletDialog - ImportWallet renders correctly 1`] = `
Array [
  <div
    className="MuiDialogContent-root HookGlobalStyles-dialogContent-6"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      common:moniker
    </p>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <div
        className="MuiInputBase-root MuiFilledInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiFilledInput-input"
          disabled={false}
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="common:wallet name"
          required={false}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="MuiBox-root MuiBox-root-7"
    />
  </div>,
  <div
    className="MuiDialogActions-root MuiDialogActions-spacing"
  >
    <div
      className="MuiBox-root MuiBox-root-8"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained HookGlobalStyles-button-1 MuiButton-containedPrimary Mui-disabled Mui-disabled"
        disabled={true}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={-1}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          common:import
        </span>
      </button>
    </div>
  </div>,
]
`;

exports[`component: CreateWalletDialog - ImportWallet renders correctly with wallet name 1`] = `
Array [
  <div
    className="MuiDialogContent-root HookGlobalStyles-dialogContent-6"
  >
    <p
      className="MuiTypography-root MuiTypography-body1"
    >
      common:moniker
    </p>
    <div
      className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
    >
      <div
        className="MuiInputBase-root MuiFilledInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiFilledInput-input"
          disabled={false}
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="common:wallet name"
          required={false}
          type="text"
          value="wallet"
        />
      </div>
    </div>
    <div
      className="MuiBox-root MuiBox-root-9"
    />
  </div>,
  <div
    className="MuiDialogActions-root MuiDialogActions-spacing"
  >
    <div
      className="MuiBox-root MuiBox-root-10"
    >
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained HookGlobalStyles-button-1 MuiButton-containedPrimary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          common:import
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
]
`;
